src/types.rs

```diff
@@ -218,6 +224,20 @@ pub enum CommercialLengthParseError {
     InvalidLength(u64),
 }
 
+/// Custom reward redemption statuses: UNFULFILLED, FULFILLED or CANCELED
+#[derive(PartialEq, serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub enum CustomRewardRedemptionStatus {
+    /// Unfulfilled reward - the user has claimed it but it is still pending.
+    #[serde(rename = "UNFULFILLED")]
+    Unfulfilled,
+    /// Fulfilled reward - the user has claimed it and the reward has been granted.
+    #[serde(rename = "FULFILLED")]
+    Fulfilled,
+    /// Cancelled reward - the reward has been cancelled before fulfillment, and any spent points have been refunded.
+    #[serde(rename = "CANCELED")]
+    Cancelled,
+}
+
```

Comment:
This is already present in https://github.com/Emilgardis/twitch_api2/blob/f93c1e2416de16bd1c2405c7ce4a4a2187677fd9/src/pubsub/channel_points.rs#L47-L65

I'd prefer having one implementation and moving the pubsub topic `channel_points` out here instead. The pubsub version is missing `Cancelled` though. Keep/merge the documentation

Reply:
The note on the above code from pubsub should be modified. I have not checked if the note is valid today.

Reply:
Hmmm, this is a tricky one. I do agree they should be the same, but given that the options are different for both - "Fulfilled/Unfulfilled/Cancelled" vs "Fulfilled/Unfulfilled/ActionTaken" (where ActionTaken is trying to represent Fulfilled/Cancelled) - I'm not sure if this is a good idea. The meaning of the values will change based on how you retrieved them (pubsub vs helix) and that feels a bit suspicious to me.

I can merge them if you still want to, and perhaps document the differences?

Reply:
keep it as is, but move it out of `types`, this seems like something where twitch has not figured out fully yet what to do.

Reply:
Done!